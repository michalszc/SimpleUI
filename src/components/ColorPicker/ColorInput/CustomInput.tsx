import React, { FC } from "react";
import styled from "styled-components";
import Styles from "../../../utils/styles";
import { IStyledTheme } from "../../../utils/theme/theme";

export interface CustomInputProps {
    prefix: string;
    value: string;
    onChange: (value: string) => void;
}

const InputContainer = styled.div<IStyledTheme>`
    display: grid;
    grid-template-columns: 22px 50px;
    color: #98A1A4;
    box-sizing: border-box;
    padding: ${ props => props.theme.sizes[3] };
    border-radius: ${ props => props.theme.sizes[1.5] };
`;

const InputPrefix = styled.div`
    background-color: #F0F0F0;
    border-top-left-radius: inherit;
    border-bottom-left-radius: inherit;
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid #e0e0e0;
    border-right: none;
    text-transform: uppercase;
    font-family: monospace;
`;

const Input = styled.input`
    color: inherit;
    font-family: monospace;
    font-size: 14px;
    line-height: 2;
    text-transform: uppercase;
    border-top-right-radius: inherit;
    border-bottom-right-radius: inherit;
    box-shadow: #F0F0F0 0px 0px 0px 1px inset;
    border: 1px solid #e0e0e0;
    border-left: none;
    outline: none;
`;

const CustomInput: FC<CustomInputProps> = ({
    prefix, value, onChange
}) => (
    <Styles>
        <InputContainer className="simpleui-colorpicker-inputcontainer">
            <InputPrefix className="simpleui-colorpicker-input-prefix">
                { prefix }
            </InputPrefix>
            <Input value={value} onChange={(e) => onChange(e.target.value)} className="simpleui-colorpicker-input"/>
        </InputContainer>
    </Styles>
);

export default CustomInput;
