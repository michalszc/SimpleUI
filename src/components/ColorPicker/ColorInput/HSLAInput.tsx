import React, { FC, useCallback, useMemo } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";
import { hexToHsla, hslaToHex } from "../../../utils";
import { floor, isNil, mapValues } from "lodash";
import Styles from "../../../utils/styles";

export interface HLSAInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

interface HSLA {
    h?: number;
    s?: number;
    l?: number;
    a?: number;
}

const HSLAContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(4, minmax(25%, 1fr));
    width: inherit;
    margin: 0 auto;
    & > .simpleui-colorpicker-inputcontainer {
        padding: ${ props => props.theme.sizes[2] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:first-of-type) {
        padding-left: ${ props => props.theme.sizes[.5] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:last-of-type) {
        padding-right: ${ props => props.theme.sizes[.5] };
    }
`;

const HSLAInput: FC<HLSAInputProps> = ({
    color, setColor
}) => {
    const hslaColor = useMemo(
        () => {
            const _hsla = hexToHsla(color);
            if (isNil(_hsla)) {
                return {
                    h: '0',
                    s: '0',
                    l: '0',
                    a: '0'
                };
            } else {
                return mapValues(_hsla, (v: number) => v.toString());
            }
        },
        [color]
    );

    const setNewColor = useCallback(
        (hsla: HSLA) => {
            const _hsla = {
                ...mapValues(hslaColor, (v: string) => Number(v)),
                ...mapValues(hsla, (v: number, k: string) => {
                    if (k === 'h') {
                        return (v > 360) ? floor(v / 10) : v;
                    } else {
                        return (v > 100) ? floor(v / 10) : v;
                    }
                }) as HSLA
            };
            setColor(
                hslaToHex(
                    _hsla.h,
                    _hsla.s,
                    _hsla.l,
                    _hsla.a
                )
            );
        }, [hslaColor]);

    return (
        <Styles>
            <HSLAContainer>
                <CustomInput prefix={'H'} value={hslaColor.h} onChange={(v: string) => setNewColor({
                    h: Number(v)
                })} />
                <CustomInput prefix={'S'} value={parseInt(hslaColor.s).toString()} onChange={(v: string) => setNewColor({
                    s: Number(v)
                })} />
                <CustomInput prefix={'L'} value={parseInt(hslaColor.l).toString()} onChange={(v: string) => setNewColor({
                    l: Number(v)
                })} />
                <CustomInput prefix={'A'} value={hslaColor.a} onChange={(v: string) => setNewColor({
                    a: Number(v)
                })} />
            </HSLAContainer>
        </Styles>
    );
};

export default HSLAInput;
