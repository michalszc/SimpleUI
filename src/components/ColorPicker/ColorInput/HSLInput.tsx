import React, { FC, useCallback, useMemo } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";
import { hexToHsl, hslToHex } from "../../../utils";
import { floor, isNil, mapValues } from "lodash";
import Styles from "../../../utils/styles";

export interface HLSInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

interface HSL {
    h?: number;
    s?: number;
    l?: number;
}

const HSLContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(3, minmax(33%, 1fr));
    width: inherit;
    margin: 0 auto;
    & > .simpleui-colorpicker-inputcontainer {
        padding: ${ props => props.theme.sizes[2] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:first-of-type) {
        padding-left: ${ props => props.theme.sizes[.5] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:last-of-type) {
        padding-right: ${ props => props.theme.sizes[.5] };
    }
`;

const HSLInput: FC<HLSInputProps> = ({
    color, setColor
}) => {
    const hslColor = useMemo(
        () => {
            const _hsl = hexToHsl(color);
            if (isNil(_hsl)) {
                return {
                    h: '0',
                    s: '0',
                    l: '0'
                };
            } else {
                return mapValues(_hsl, (v: number) => v.toString());
            }
        },
        [color]
    );

    const setNewColor = useCallback(
        (hsl: HSL) => {
            const _hsl = {
                ...mapValues(hslColor, (v: string) => Number(v)),
                ...mapValues(hsl, (v: number, k: string) => {
                    if (k === 'h') {
                        return (v > 360) ? floor(v / 10) : v;
                    } else {
                        return (v > 100) ? floor(v / 10) : v;
                    }
                }) as HSL
            };
            setColor(
                hslToHex(
                    _hsl.h,
                    _hsl.s,
                    _hsl.l
                )
            );
        }, [hslColor]);

    return (
        <Styles>
            <HSLContainer className={"simpleui-colorpicker-hslcontainer"}>
                <CustomInput prefix={'H'} value={hslColor.h} onChange={(v: string) => setNewColor({
                    h: Number(v)
                })} />
                <CustomInput prefix={'S'} value={parseInt(hslColor.s).toString()} onChange={(v: string) => setNewColor({
                    s: Number(v)
                })} />
                <CustomInput prefix={'L'} value={parseInt(hslColor.l).toString()} onChange={(v: string) => setNewColor({
                    l: Number(v)
                })} />
            </HSLContainer>
        </Styles>
    );
};

export default HSLInput;
