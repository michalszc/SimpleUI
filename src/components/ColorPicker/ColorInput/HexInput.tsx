import React, { FC } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";

export interface HexInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

const HexContainer = styled.div`
    display: grid;
    justify-content: center;
    & > .simpleui-colorpicker-inputcontainer {
        grid-template-columns: 22px 70px;
    }
`;

const HexInput: FC<HexInputProps> = ({
    color, setColor
}) => (
    <HexContainer>
        <CustomInput prefix={'#'} value={color.replace('#', '')} onChange={(value: string) => setColor(`#${value}`)} />
    </HexContainer>
);

export default HexInput;
