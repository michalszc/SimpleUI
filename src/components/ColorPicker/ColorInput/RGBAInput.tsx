import React, { FC, useCallback, useMemo } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";
import { hexToRgba, rgbaToHex } from "../../../utils";
import { floor, isNil, mapValues } from "lodash";
import Styles from "../../../utils/styles";

export interface RGBAInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

interface RGBA {
    r?: number;
    g?: number;
    b?: number;
    a?: number;
}

const RGBAContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(4, minmax(25%, 1fr));
    width: inherit;
    margin: 0 auto;
    & > .simpleui-colorpicker-inputcontainer {
        padding: ${ props => props.theme.sizes[2] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:first-of-type) {
        padding-left: ${ props => props.theme.sizes[.5] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:last-of-type) {
        padding-right: ${ props => props.theme.sizes[.5] };
    }
`;

const RGBAInput: FC<RGBAInputProps> = ({
    color, setColor
}) => {
    const rgbaColor = useMemo(
        () => {
            const _rgba = hexToRgba(color);
            if (isNil(_rgba)) {
                return {
                    r: '0',
                    g: '0',
                    b: '0',
                    a: '0'
                };
            } else {
                return mapValues(_rgba, (v: number) => v.toString());
            }
        },
        [color]
    );

    const setNewColor = useCallback(
        (rgba: RGBA) => {
            const _rgba = {
                ...mapValues(rgbaColor, (v: string) => Number(v)),
                ...mapValues(rgba, (v: number, k: string) => {
                    if (k !== 'a') {
                        return (v > 255) ? floor(v / 10) : v;
                    } else {
                        return (v > 100) ? floor(v / 10) : v;
                    }
                }) as RGBA
            };
            setColor(
                rgbaToHex(
                    _rgba.r,
                    _rgba.g,
                    _rgba.b,
                    _rgba.a
                )
            );
        }, [rgbaColor]);

    return (
        <Styles>
            <RGBAContainer className={"simpleui-colorpicker-rgbacontainer"}>
                <CustomInput prefix={'R'} value={rgbaColor.r} onChange={(v: string) => setNewColor({
                    r: Number(v)
                })} />
                <CustomInput prefix={'G'} value={rgbaColor.g} onChange={(v: string) => setNewColor({
                    g: Number(v)
                })} />
                <CustomInput prefix={'B'} value={rgbaColor.b} onChange={(v: string) => setNewColor({
                    b: Number(v)
                })} />
                <CustomInput prefix={'A'} value={rgbaColor.a} onChange={(v: string) => setNewColor({
                    a: Number(v)
                })} />
            </RGBAContainer>
        </Styles>
    );
};

export default RGBAInput;
