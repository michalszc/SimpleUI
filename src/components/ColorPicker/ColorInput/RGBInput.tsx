import React, { FC, useCallback, useMemo } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";
import { hexToRgb, rgbToHex } from "../../../utils";
import { floor, isNil, mapValues } from "lodash";
import Styles from "../../../utils/styles";

export interface RGBInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

interface RGB {
    r?: number;
    g?: number;
    b?: number;
}

const RGBContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(3, minmax(33%, 1fr));
    width: inherit;
    margin: 0 auto;
    & > .simpleui-colorpicker-inputcontainer {
        padding: ${ props => props.theme.sizes[2] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:first-of-type) {
        padding-left: ${ props => props.theme.sizes[.5] };
    }
    & > .simpleui-colorpicker-inputcontainer:not(:last-of-type) {
        padding-right: ${ props => props.theme.sizes[.5] };
    }
`;

const RGBAInput: FC<RGBInputProps> = ({
    color, setColor
}) => {
    const rgbColor = useMemo(
        () => {
            const _rgb = hexToRgb(color);
            if (isNil(_rgb)) {
                return {
                    r: '0',
                    g: '0',
                    b: '0'
                };
            } else {
                return mapValues(_rgb, (v: number) => v.toString());
            }
        },
        [color]
    );

    const setNewColor = useCallback(
        (rgb: RGB) => {
            const _rgb = {
                ...mapValues(rgbColor, (v: string) => Number(v)),
                ...mapValues(rgb, (v: number) => 
                    (v > 255) ? floor(v / 10) : v
                ) as RGB
            };
            setColor(
                rgbToHex(
                    _rgb.r,
                    _rgb.g,
                    _rgb.b,
                )
            );
        }, [rgbColor]);

    return (
        <Styles>
            <RGBContainer>
                <CustomInput prefix={'R'} value={rgbColor.r} onChange={(v: string) => setNewColor({
                    r: Number(v)
                })} />
                <CustomInput prefix={'G'} value={rgbColor.g} onChange={(v: string) => setNewColor({
                    g: Number(v)
                })} />
                <CustomInput prefix={'B'} value={rgbColor.b} onChange={(v: string) => setNewColor({
                    b: Number(v)
                })} />
            </RGBContainer>
        </Styles>
    );
};

export default RGBAInput;
