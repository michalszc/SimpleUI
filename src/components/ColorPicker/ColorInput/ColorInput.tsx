import React, { FC } from "react";
import * as CSS from "csstype";
import HexInput from "./HexInput";
import RGBInput from "./RGBInput";

export interface ColorInputProps {
    type: 'hex' | 'hsl' | 'hsla' | 'rgb' | 'rgba';
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

const ColorInput: FC<ColorInputProps> = ({
    type, color, setColor
}) => {
    if (type === 'hex') {
        return (
            <HexInput color={color} setColor={setColor} />
        );
    }

    if (type === 'rgb') {
        return (
            <RGBInput color={color} setColor={setColor} />
        );
    }

    return (
        <></>
    );
}

export default ColorInput;
