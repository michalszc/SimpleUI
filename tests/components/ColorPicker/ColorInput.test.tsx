import React from "react";
import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import ColorInput from "../../../src/components/ColorPicker/ColorInput/ColorInput";

describe("ColorInput", () => {
    const color = '#FFF';
    const mockFn = jest.fn();
    test("should properly render ColorInput component - type hex", () => {
        const { container, getByRole } = render(<ColorInput color={color} setColor={mockFn} type={'hex'}/>);
        const input = getByRole('textbox');
        expect(input).toHaveValue('FFF');
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorInput component - type hsl", () => {
        const { container, getAllByRole } = render(<ColorInput color={color} setColor={mockFn} type={'hsl'}/>);
        const inputs = getAllByRole('textbox');
        expect(inputs[0]).toHaveValue('0');
        expect(inputs[1]).toHaveValue('0');
        expect(inputs[2]).toHaveValue('100');
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorInput component - type hsla", () => {
        const { container, getAllByRole } = render(<ColorInput color={color} setColor={mockFn} type={'hsla'}/>);
        const inputs = getAllByRole('textbox');
        expect(inputs[0]).toHaveValue('0');
        expect(inputs[1]).toHaveValue('0');
        expect(inputs[2]).toHaveValue('100');
        expect(inputs[3]).toHaveValue('100');
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorInput component - type rgb", () => {
        const { container, getAllByRole } = render(<ColorInput color={color} setColor={mockFn} type={'rgb'}/>);
        const inputs = getAllByRole('textbox');
        expect(inputs[0]).toHaveValue('255');
        expect(inputs[1]).toHaveValue('255');
        expect(inputs[2]).toHaveValue('255');
        expect(container).toMatchSnapshot();
    });
    test("should properly render ColorInput component - type rgba", () => {
        const { container, getAllByRole } = render(<ColorInput color={color} setColor={mockFn} type={'rgba'}/>);
        const inputs = getAllByRole('textbox');
        expect(inputs[0]).toHaveValue('255');
        expect(inputs[1]).toHaveValue('255');
        expect(inputs[2]).toHaveValue('255');
        expect(inputs[3]).toHaveValue('100');
        expect(container).toMatchSnapshot();
    });
});
