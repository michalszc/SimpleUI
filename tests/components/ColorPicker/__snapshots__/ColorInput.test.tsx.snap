// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput should properly render ColorInput component - type hex 1`] = `
<div>
  <div
    class="sc-hLBbgP hbVVJF"
  >
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        #
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="FFF"
      />
    </div>
  </div>
</div>
`;

exports[`ColorInput should properly render ColorInput component - type hsl 1`] = `
<div>
  <div
    class="sc-iBYQkv boYHRO"
  >
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        H
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="0"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        S
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="0"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        L
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="100"
      />
    </div>
  </div>
</div>
`;

exports[`ColorInput should properly render ColorInput component - type hsla 1`] = `
<div>
  <div
    class="sc-gKPRtg hDVxWN"
  >
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        H
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="0"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        S
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="0"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        L
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="100"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        A
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="100"
      />
    </div>
  </div>
</div>
`;

exports[`ColorInput should properly render ColorInput component - type rgb 1`] = `
<div>
  <div
    class="sc-eDvSVe eJvcwf"
  >
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        R
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        G
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        B
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
  </div>
</div>
`;

exports[`ColorInput should properly render ColorInput component - type rgba 1`] = `
<div>
  <div
    class="sc-jSUZER ggHwA-d"
  >
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        R
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        G
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        B
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="255"
      />
    </div>
    <div
      class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
    >
      <div
        class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
      >
        A
      </div>
      <input
        class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
        value="100"
      />
    </div>
  </div>
</div>
`;
